diff --git a/zml/tensor.zig b/zml/tensor.zig
index 07c887d0..b4edf9b6 100644
--- a/zml/tensor.zig
+++ b/zml/tensor.zig
@@ -880,7 +880,7 @@ pub const Tensor = struct {
         return _result(self._shape, op.result(0));
     }
 
-    fn convolution(self: Tensor, other: Tensor, opts: dialects.stablehlo.ConvolutionOpts) Tensor {
+    pub fn convolution(self: Tensor, other: Tensor, opts: dialects.stablehlo.ConvolutionOpts) Tensor {
         stdx.debug.assert(self.rank() == other.rank(), "convolution expects tensor ranks to match, got {} and {}", .{ self.rank(), other.rank() });
         const N = self.rank();
         stdx.debug.guard(opts.window_strides.len == N - 2, @src());
